frontend/packages/integration-tests-cypress/tests/storage/snapshot.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { detailsPage } from '../../views/details-page';
+import { submitButton } from '../../views/form';
+
+const snapshotName = `${PVC.metadata.name}-snapshot`;
+
+if (Cypress.env('BRIDGE_AWS')) {
+  describe('Snapshot Tests', () => {
+    before(() => {
+      cy.login();
+      cy.createProject(testName);
+      cy.exec(`echo '${JSON.stringify(PVC)}' | oc apply -n ${testName} -f -`);
+      cy.exec(`echo '${JSON.stringify(testerDeployment)}' | oc apply -n ${testName} -f -`);
+      cy.exec(`echo '${JSON.stringify(SnapshotClass)}' | oc apply -f -`);
+      cy.clickNavLink(['Storage', 'Persistent Volume Claims']);
+      listPage.filter.byName(PVC.metadata.name);
+      // Let the PVC get bound
+      cy.contains(status, 'Bound');
+    });
+
+    afterEach(() => {
+      checkErrors();
+    });
+
+    after(() => {
+      cy.exec(`echo '${JSON.stringify(testerDeployment)}' | oc delete -n ${testName} -f -`);
+      cy.exec(`echo '${JSON.stringify(PVC)}' | oc delete -n ${testName} -f -`);
+      cy.exec(`echo '${JSON.stringify(SnapshotClass)}' | oc delete -f -`);
+      cy.deleteProject(testName);
+      cy.logout();
+    });
+
+    it('Creates Snapshot', () => {
+      cy.clickNavLink(['Volume Snapshots']);
+      listPage.clickCreateYAMLbutton();
+      cy.byTestID(pvcDropdown).click();
+      cy.get(dropdownFirstItem)
+        .first()
+        .click();
+      cy.byTestID(snapshotClassDropdown).click();
+      cy.get(dropdownFirstItem)
+        .first()
+        .click();
+      cy.get(submitButton).click();
+      cy.location('pathname').should(
+        'include',
+        `snapshot.storage.k8s.io~v1beta1~VolumeSnapshot/${PVC.metadata.name}-snapshot`,
+      );
+      detailsPage.titleShouldContain(PVC.metadata.name);
+      cy.contains(status, 'Ready');
+      return cy
+        .exec(`oc get VolumeSnapshot ${PVC.metadata.name}-snapshot -n ${testName} -o json`)
+        .its('stdout')
+        .then((res) => {
+          const volumeSnapshot = JSON.parse(res);
+          cy.get(SnapshotDetails.name).contains(volumeSnapshot.metadata.name);
+          cy.get(SnapshotDetails.namespace).contains(volumeSnapshot.metadata.namespace);
+          cy.get(SnapshotDetails.vsc).contains(
+            volumeSnapshot.status.boundVolumeSnapshotContentName,
+          );
+          cy.get(SnapshotDetails.sc).contains(volumeSnapshot.spec.volumeSnapshotClassName);
+          cy.get(SnapshotDetails.pvc).contains(
+            volumeSnapshot.spec.source.persistentVolumeClaimName,
+          );
+        });
+    });
+
+    it('Lists Snapshot', () => {
+      cy.clickNavLink(['Volume Snapshots']);
+      listPage.rows.shouldBeLoaded();
+      listPage.rows.shouldExist(snapshotName);
+      listPage.rows.shouldNotExist(`${snapshotName}dup`);
+    });
+
+    it('Deletes Snapshot', () => {
+      listPage.rows.clickKebabAction(snapshotName, 'Delete Volume Snapshot');
+      cy.get('#confirm-action').click();
```

Comment:
Can we add a `data-test` prop to this? Also, would `modal.submit();` work in this case?  Might want to also see `modal.shouldBeOpened() && modal.shouldBeClosed()` helper methods.

Reply:
modal.submit() works!